include/aspect/postprocess/geoid.h

```diff
@@ -0,0 +1,347 @@
+
+
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+
+        /**
+         * Get a reference to the multipole expansion of the potential
+         * at the surface.
+         */
+        const internal::ExternalMultipoleExpansion<dim> &
+        get_surface_potential_expansion();
+
+        /**
+         * Get a reference to the multipole expansion of the potential
+         * at the CMB.
+         */
+        const internal::InternalMultipoleExpansion<dim> &
+        get_cmb_potential_expansion();
+
+        /**
+         * Let the postprocessor manager know about the other postprocessors
+         * which this one depends on.  Specifically, BoundaryPressures and
+         * BoundaryDensities.
+         */
+        virtual
+        std::list<std::string>
+        required_other_postprocessors() const;
+
+      private:
+
+        /**
+         * Compute the multipole expansion of the internal density structure.
+         */
+        void compute_internal_density_expansions();
+
+        /**
+         * Compute the harmonic expansion of the dynamic topography on the
+         * top and bottom boundaries.
+         */
+        void compute_topography_expansions();
+
+        /**
+         * Compute the geoid at the top and bottom of the domain.  This has
+         * contributions from internal density structure, the bottom topography,
+         * and the top topography.
+         */
+        void compute_geoid_expansions();
+
+        /**
+         * Write the geoid and associated information to an output file.
+         */
+        void output_geoid_information();
+
+        /**
+         * A parameter that we read from the input file that denotes whether
+         * we should include the contribution by dynamic topography.
+         */
+        bool include_topography_contribution;
+
+        /**
+         * The density below the bottom boundary (typically the density of
+         * liquid iron).
+         */
+        double density_below;
+
+        /**
+         * The density above the top boundary (typically the density of
+         * air or water).
+         */
+        double density_above;
+
+        /**
+         * The maximum harmonic degree for the expansion.
+         */
+        unsigned int max_degree;
+
+        /**
+         * The inner radius of the planet
+         */
+        double inner_radius;
+
+        /**
+         * The outer radius of the planet
+         */
+        double outer_radius;
+
+        /**
+         * A pointer to the postprocessor for computing boundary
+         * pressures.
+         */
+        BoundaryPressures<dim> *boundary_pressure_postprocessor;
```

Comment:
As far as I can see you only use this in `compute_topography_expansions()` and update it in every timestep anyway. Would it make sense to make it a local variable in that function? This would reduce the number of members of this class.